### PR TITLE
[SCOUTCLOUD] initial

### PR DESCRIPTION
**Pull request by sevenzing**
This PR is initial creation of new `scoutcloud` service.

### EPIC
https://github.com/blockscout/blockscout-rs/issues/801

### Important for code review

1. this is first service where we decided to merge `server` and `logic` crate into one
2. this PR contains initialization of service and github client feature
3. for github client I added mocked server implementation. It uses saved (see `fetch.py`) responses from real github API
4. there is code for `auth` logic, but I think this logic will be changed during next PRs
5.

### CODE REVIEW

File: scoutcloud/scoutcloud/src/logic/github/mod.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    #[cfg(test)]
+    pub fn from_mock(mock: &MockedGithubRepo) -> Result<Self, octocrab::Error> {
+        Self::new(
+            mock.token.clone(),
+            mock.owner.clone(),
+            mock.repo.clone(),
+            Some(mock.default_main_branch.clone()),
+            Some(mock.server.base_url().as_str()),
+        )
+    }
```
Comment (rimrakhimov) [inline scoutcloud/scoutcloud/src/logic/github/mod.rs:59]: Nit: may be extracted into `TryFrom` trait

File: scoutcloud/scoutcloud/src/logic/github/types.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+impl CreateBlobRequest {
+    pub fn with_default_encoding(content: impl Display) -> Self {
+        Self {
+            content: content.to_string(),
+            encoding: "utf-8".to_string(),
+        }
+    }
+}
+
+pub type BlobResponse = OnlySha;
```
Comment (rimrakhimov) [inline scoutcloud/scoutcloud/src/logic/github/types.rs:24]: Why not `CreateBlobResponse`?

File: scoutcloud/scoutcloud/src/logic/github/types.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TreeItem {
+    pub path: String,
+    pub mode: String,
+    #[serde(rename = "type")]
+    pub _type: String,
+    pub sha: String,
+}
+
+pub type TreeResponse = OnlySha;
```
Comment (rimrakhimov) [inline scoutcloud/scoutcloud/src/logic/github/types.rs:57]: Why not `CreateTreeResponse`?

File: scoutcloud/scoutcloud/src/server/settings.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub struct DatabaseSettings {
+    pub url: String,
+    #[serde(default)]
+    pub create_database: bool,
+    #[serde(default)]
+    pub run_migrations: bool,
+}
```
Comment (rimrakhimov) [inline scoutcloud/scoutcloud/src/server/settings.rs:34]: We have `DatabaseSettings` defined in blockscout-service-launcher. Maybe use that one?

**Review by rimrakhimov — APPROVED** [review]

File: scoutcloud/scoutcloud-proto/proto/v1/api_config_http.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    - selector: blockscout.scoutcloud.v1.Scoutcloud.UpdateConfigPartial
+      patch: /api/v1/instances/{instance_id}/config
+      body: "*"
+
+    - selector: blockscout.scoutcloud.v1.Scoutcloud.UpdateInstanceStatus
+      post: /api/v1/instances/{instance_id}/status:update
+      body: "*"
+
+    - selector: blockscout.scoutcloud.v1.Scoutcloud.GetDeployment
+      get: /api/v1/deployments/{deployment_id}
```
[the remark this answers is not in the document]
Comment (sevenzing) [reply scoutcloud/scoutcloud-proto/proto/v1/api_config_http.yaml:29]: yeah, UUID

[the remark this answers is not in the document]
Comment (sevenzing) [reply scoutcloud/scoutcloud-proto/proto/v1/api_config_http.yaml:29]: (forget to add external_id for deployment btw, will add it)

**Pull request merged**